### ISSUE TITLE
Add Setter for XCMP QueueConfigData

### ISSUE THREAD

**Issue by apopiak** [paritytech/cumulus#801]
https://github.com/paritytech/cumulus/blob/8cc008b6a9f70345ec438c4f802c12bbaa05dbd7/pallets/xcmp-queue/src/lib.rs#L166-L168

QueueConfigData is read but there is no way to write it (in contrast to e.g. Polkadot's HostConfiguration).
Would be good to add a setter (that is only available to root).

**Comment by Doordashcon** [comment]
Hi @apopiak so we need a separate function exclusively for inserting data(`QueueConfigData`) into the storage item `QueueConfig`?

**Comment by Doordashcon** [comment]
Nice! The root/origin would be `type QueueConfigDataOrigin:  EnsureOrigin<Self::Origin>;` enabling implementors assign governance?

**Comment by apopiak** [comment]
yeah, sure, we could introduce `ConfigUpdateOrigin` or `UpdateOrigin` or sth.
Though, just requiring root would allow us to avoid downstream changes.
Either works, though.

**Comment by Doordashcon** [comment]
I'm on it!

**Comment by KiChjang** [comment]
@Doordashcon Since it involves a storage write, I think using `T::DbWeight::get().writes(1)` is good enough.

**Comment by Doordashcon** [comment]
> @Doordashcon Since it involves a storage write, I think using `T::DbWeight::get().writes(1)` is good enough.

won't I need to `kill` existing storage if any before the update?

**Comment by KiChjang** [comment]
No? You're overwriting the existing `QueueConfigData` with a new one.

**Comment by KiChjang** [comment]
Fixed in #931, weights for the pallet is tracked in #961.
